torch/fx/node.py

```diff
@@ -33,6 +33,32 @@ def __init__(self, graph: 'Graph', name: str, op: str, target: Target,
         self.args = args
         self.kwargs = kwargs
         self.uses = 0
+        self._prev = self
+        self._next = self
+        self._erased = False
+
+    @property
+    def next(self) -> 'Node':
+        return self._next
+
+    @property
+    def prev(self) -> 'Node':
+        return self._prev
+
+    def prepend(self, x: 'Node'):
+        assert self.graph == x.graph, "Attempting to move a Node into a different Graph"
+        x._remove_from_list()
+        # current: p -> self
+        # after p -> x -> self
+        p = self._prev
+        p._next, x._prev, x._next, self._prev = x, p, self, x
+
+    def append(self, x: 'Node'):
+        self._next.prepend(x)
+
+    def _remove_from_list(self):
```

Comment:
the user never should see nodes that exist but are not in the list somewhere. The public function `erase_node` is the public API. This one doesn't set the _erased attribute because it also used when moving a node.